index.d.ts

```diff
@@ -83,9 +84,9 @@ export declare interface Tracer extends opentracing.Tracer {
    * unless there is already an active span or `childOf` option. Note that this
    * option is deprecated and has been removed in version 4.0.
    */
-  trace<T> (name: string, fn: (span: Span) => T): T;
-  trace<T> (name: string, fn: (span: Span, done: (error?: Error) => void) => T): T;
-  trace<T> (name: string, options: TraceOptions & SpanOptions, fn: (span?: Span, done?: (error?: Error) => void) => T): T;
+  trace<T> (name: string, fn: (span?: tracer.Span, fn?: (error?: Error) => any) => T): T;
```

Comment:
This was split into two signatures both of which _require_ that span is handled.